pkgs/desktops/lomiri/development/lomiri-ui-toolkit/default.nix

```diff
@@ -0,0 +1,179 @@
+# TODO
+# - tests
+# - check if there's a better solution for all the patching
+# - "localisation through gettext" is broken
+#   - if multiple QML modules from different packages use this API and get used in the same application,
+#     the textdomain will be wrong for at least 1 of them
+#   - lomiri-system-settings has an awkward workaround, but we should try to fix this in here instead
+{ stdenv
+, lib
+, fetchFromGitLab
+, gitUpdater
+, testers
+, dbus-test-runner
+, dpkg
+, gdb
+, glib
+, lttng-ust
+, perl
+, pkg-config
+, python3
+, qmake
+, qtbase
+, qtdeclarative
+, qtfeedback
+, qtgraphicaleffects
+, qtpim
+, qtquickcontrols2
+, qtsystems
+, ubuntu-themes
+, wrapQtAppsHook
+, xvfb-run
+}:
+
+let
+  listToQtVar = list: suffix: lib.strings.concatMapStringsSep ":" (drv: "${lib.getBin drv}/${suffix}") list;
```

Comment:
`listToQtVar = suffix: lib.makeSearchPathOutput "bin" suffix;`

This should be simpler, haven't tested

Reply:
Produces same paths, thanks!